### ISSUE TITLE
ESLint couldn't find the plugin "eslint-plugin-react-hook".

### ISSUE THREAD

**Issue by adamklepacz** [eslint/eslint#12637]
**Tell us about your environment**

* **ESLint Version:**   "eslint": "6.6.0",
* **Node Version:** v10.15.3
* **npm Version:** 6.12.0

**What parser (default, Babel-ESLint, etc.) are you using?**
```
    "@babel/core": "7.7.2",
    "@babel/plugin-proposal-class-properties": "7.7.0",
    "@babel/preset-env": "7.7.1",
    "@babel/preset-react": "7.7.0",
    "babel-eslint": "10.0.3",
    "babel-jest": "24.9.0",
    "babel-loader": "8.0.6",
    "babel-plugin-jsx-remove-data-test-id": "2.0.1",
```

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "env": {
    "browser": true,
    "jest": true
  },
  "extends": ["airbnb", "plugin:prettier/recommended", "plugin:react-hooks"],
  "parser": "babel-eslint",
  "settings": {
    "import/resolver": {
      "node": {},
      "webpack": {
        "config": "webpack.config.js"
      }
    }
  },
  "rules": {
    "array-bracket-spacing": ["error", "never"],
    "arrow-parens": ["error", "as-needed"],
    "arrow-spacing": ["error", { "before": true, "after": true }],
    "block-spacing": ["warn"],
    "brace-style": ["error", "1tbs", { "allowSingleLine": true }],
    "comma-dangle": ["error", "never"],
    "comma-spacing": ["warn"],
    "computed-property-spacing": ["error", "never"],
    "consistent-return": ["off"],
    "eqeqeq": ["error", "smart"],
    "func-call-spacing": ["error", "never"],
    "func-names": ["error", "never"],
    "import/extensions": "error",
    "import/first": "error",
    "import/newline-after-import": "error",
    "import/no-cycle": "error",
    "import/no-duplicates": "error",
    "import/no-extraneous-dependencies": ["error", { "devDependencies": true }],
    "import/no-named-default": "error",
    "import/no-unresolved": "error",
    "import/prefer-default-export": "off",
    "jsx-a11y/click-events-have-key-events": "off",
    "jsx-a11y/label-has-associated-control": 0,
    "jsx-a11y/label-has-for": 0,
    "jsx-a11y/no-noninteractive-element-interactions": "off",
    "jsx-a11y/no-static-element-interactions": "off",
    "key-spacing": "error",
    "keyword-spacing": ["error", { "before": true, "after": true }],
    "newline-before-return": ["error"],
    "no-console": "error",
    "no-debugger": ["warn"],
    "no-else-return": "error",
    "no-mixed-spaces-and-tabs": ["error", "smart-tabs"],
    "no-multi-spaces": ["error", { "ignoreEOLComments": false }],
    "no-multiple-empty-lines": ["error", { "max": 1 }],
    "no-plusplus": ["error", { "allowForLoopAfterthoughts": true }],
    "no-shadow": "off",
    "no-underscore-dangle": "off",
    "no-unused-expressions": "off",
    "no-unused-vars": [
      "warn",
      { "vars": "local", "args": "none", "ignoreRestSiblings": true }
    ],
    "no-use-before-define": "error",
    "no-var": "error",
    "object-curly-spacing": ["error", "always"],
    "prefer-arrow-callback": "error",
    "prefer-const": ["error", { "destructuring": "all" }],
    "prefer-destructuring": "error",
    "quote-props": ["error", "as-needed"],
    "quotes": ["error", "single", { "avoidEscape": true }],
    "semi": ["warn", "always"],
    "space-before-blocks": "error",
    "space-before-function-paren": [
      "error",
      {
        "anonymous": "never",
        "named": "never",
        "asyncArrow": "always"
      }
    ],
    "space-in-parens": ["error", "never"],
    "space-infix-ops": "error",
    "import/default": "error",
    "import/named": "error",
    "import/namespace": "error",
    "import/order": [
      "error",
      {
        "groups": [
          ["builtin", "external"],
          ["index", "sibling", "parent", "internal"]
        ],
        "newlines-between": "always"
      }
    ],
    "import/no-useless-path-segments": "error",
    "react/forbid-prop-types": 0,
    "react/jsx-filename-extension": 0,
    "react/jsx-props-no-spreading": 0,
    "react/require-default-props": 0,
    "react/state-in-constructor": 0,
    "react-hooks/rules-of-hooks": "error",
    "react-hooks/exhaustive-deps": "warn"
  },
  "globals": {
    "_": false,
    "window": false,
    "fetch": false,
    "document": false
  },
  "overrides": [
    {
      "files": ["client/tests/__mocks__/*.*"],
      "rules": {
        "react/prop-types": "off"
      }
    }
  ]
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I've been trying to add `eslint-plugin-react-hooks` to my application. I've read that current airbnb config doesn't support the latest version of `eslint-plugin-react-hooks` package, so I've installed recommended by AirBnB, v.1.7.0.  At this moment I dont get any error after instalation of 1.7.0 version. But when I run `eslint ./ --ext \".js,.jsx\" --max-warnings=0 --fix` I'm getting: 

```js
Oops! Something went wrong! :(

ESLint: 6.6.0.

ESLint couldn't find the plugin "eslint-plugin-react-hook".

(The package "eslint-plugin-react-hook" was not found when loaded as a Node module from the directory "/Users/adamklepacz/Documents/projects/eoc".)

It's likely that the plugin isn't installed correctly. Try reinstalling by running the following:

    npm install eslint-plugin-react-hook@latest --save-dev

The plugin "eslint-plugin-react-hook" was referenced from the config file in ".eslintrc".
```

Here as you can see, eslint yields at me that plugin `eslint-plugin-react-hook` (withous 's' at the end) is missing. Such a package doesn't even exist in the npm library so I think it is a typo in that particular error or something else, but I can't figure it out.

**What did you expect to happen?**
I expect ESlint to run as expected and lint all the errors including those two extra eslint rules:
```
    "react-hooks/rules-of-hooks": "error",
    "react-hooks/exhaustive-deps": "warn"
```

**What actually happened? Please include the actual, raw output from ESLint.**
Actual input is like this:

```
Oops! Something went wrong! :(

ESLint: 6.6.0.

ESLint couldn't find the plugin "eslint-plugin-react-hook".

(The package "eslint-plugin-react-hook" was not found when loaded as a Node module from the directory "/Users/adamklepacz/Documents/projects/eoc".)

It's likely that the plugin isn't installed correctly. Try reinstalling by running the following:

    npm install eslint-plugin-react-hook@latest --save-dev

The plugin "eslint-plugin-react-hook" was referenced from the config file in ".eslintrc".

If you still can't figure out the problem, please stop by https://gitter.im/eslint/eslint to chat with the team.

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! eoc-app@0.0.1 lint: `eslint ./ --ext ".js,.jsx" --max-warnings=0`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the eoc-app@0.0.1 lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/adamklepacz/.npm/_logs/2019-12-04T06_53_24_016Z-debug.log
ERROR: "lint" exited with 2.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! eoc-app@0.0.1 check: `npm-run-all lint format stylelint`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the eoc-app@0.0.1 check script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/adamklepacz/.npm/_logs/2019-12-04T06_53_24_050Z-debug.log
```

**Are you willing to submit a pull request to fix this bug?**
Basically not, because I don't know what happend.

**Comment by yeonjuan** [comment]
@adamklepacz 
Can you try this? Maybe there is no sharable configuration in plugin "react-hooks".
1. Delete  "plugin:react-hooks" in "extends".
2. Add "react-hooks" in "plugins".
```json
{
  "extends": ["airbnb", "plugin:prettier/recommended"],
  "plugins": ["react-hooks"],
}
```

**Comment by kaicataldo** [comment]
To follow up on what @yeonjuan mentioned, you can find the documentation on how to use the plugin [here](https://www.npmjs.com/package/eslint-plugin-react-hooks).

**Comment by adamklepacz** [comment]
Thank you guys, @yeonjuan helped me a lot :)
